cocos/core/assets/texture-base.jsb.ts

```diff
@@ -200,6 +200,15 @@ textureBaseProto.getSamplerInfo = function () {
     return this._samplerInfo;
 };
 
+const oldDestroy = textureBaseProto.destroy;
+textureBaseProto.destroy = function () {
+    let destroyed = oldDestroy.call(this);
+    if (destroyed && legacyCC.director.root?.batcher2D) {
+        legacyCC.director.root.batcher2D._releaseDescriptorSetCache(this._textureHash);
```

Comment:
It's safer to use `this.getHash()` instead of `this._textureHash`.

Refer to texture-base.jsb.ts
```ts
const oldGetHash = textureBaseProto.getHash;
textureBaseProto.getHash = function () {
    if (this._textureHash === 0) { // lazy init here
        this._textureHash = oldGetHash.call(this); // JSB method
    }
    return this._textureHash;
};

```